src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,92 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private final Map<Class<?>, Object> components = new HashMap<>();
+
+    private Injector() {
+        try {
+            registerComponents();
+            injectDependencies();
+        } catch (Exception e) {
+            throw new RuntimeException("Error initializing injector", e);
```

Comment:
It is better to replace many exceptions that have a common parent with a general parental exception. In this case, you can catch `ReflectiveOperationException` instead of `Exception` to be more specific about what you are catching.